src/preact/compat.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as compat from /*OK*/ 'preact/compat';
+
+/**
+ * @param {function(JsonObject, {current: (T|null)}):PreactDef.Renderable} fn
+ * @return {PreactDef.Renderable}
+ * @template T
+ */
+export function forwardRef(fn) {
```

Comment:
I'd rather keep this in our `src/preact/index.js`. I think it's definitely not a compatibility layer for us. And I think Preact is changing their mind on this as well.

Also, we need to expose `useImperativeHandle` as well.

Reply:
There's a caveat with the types for `useImperativeHandle`, see [this comment](https://github.com/ampproject/amphtml/pull/29486/files#r460187444)